### DISCUSSION TITLE
Language Server Taking More Memoery

### DISCUSSION THREAD

**Discussion by LavanyaBurlagadda** [TypeFox/monaco-languageclient#462] [Q&A]
Hi @kaisalmen 
 
I tried integrating language Server with Monaco Editor. I have two issues
1 - Getting suggestions lately
2 - Language server is taking almost 1GB data, even if I use `"-Xmx512m"` 

```
  const args = [
    "-Declipse.application=org.eclipse.jdt.ls.core.id1",
    "-Dosgi.bundles.defaultStartLevel=4",
    "-Declipse.product=org.eclipse.jdt.ls.core.product",
    "-Dlog.level=ALL",
    "-noverify",
    "-XX:MinHeapFreeRatio=5",
    "-XX:MaxHeapFreeRatio=10",
    "-XX:GCTimeRatio=4",
    "-XX:AdaptiveSizePolicyWeight=90",
    "-Dsun.zip.disableMemoryMapping=true",
    "-Xmx512m",
    "-Xms100m",
    "-jar",
    `${jdtPath}/jdt-language-server-1.9.0-202203031534/plugins/org.eclipse.equinox.launcher_1.6.400.v20210924-0641.jar`,
    "-configuration",
    `${jdtPath}/jdt-language-server-1.9.0-202203031534/config_linux`,
    "-data",
    `${dataPath}`,
    "--add-modules=ALL-SYSTEM",
    "--add-opens java.base/java.util=ALL-UNNAMED",
    "--add-opens java.base/java.lang=ALL-UNNAMED",
  ];
```

Is there anything that I am missing?

_Originally posted by @LavanyaBurlagadda in https://github.com/TypeFox/monaco-languageclient/issues/299#issuecomment-1438170341_

**Comment by kaisalmen** [comment]
> 1 - Getting suggestions lately

Hi @LavanyaBurlagadda what is still not clear to me: Did it work already or did it never work? Did you do it similar too this? https://github.com/TypeFox/monaco-languageclient/issues/449#issuecomment-1373964059
Are you able to share your code / repo?

> 2 - Language server is taking almost 1GB data, even if I use "-Xmx512m"

We don't provide support for jdt.ls here.

**Comment by LavanyaBurlagadda** [comment]
@kaisalmen 

`Hi @LavanyaBurlagadda what is still not clear to me: Did it work already or did it never work? Did you do it similar too this? https://github.com/TypeFox/monaco-languageclient/issues/449#issuecomment-1373964059`


Currently we are running the Language Server in the Docker container.

When I am checking in the suggestions for Language Server in Local System(Docker is running in local system) very quickly less than 4 seconds.

But when I am trying to run the docker in server side we are getting suggestions lately almost it is taking more than 1 minute. 

I am not sure what is the issue exactly whether the issue is with the available RAM or CORES

`We don't provide support for jdt.ls here.`

Currently we are using [this](https://github.com/eclipse/eclipse.jdt.ls) for Language Server

Are you referring to this only https://github.com/eclipse/eclipse.jdt.ls?
Regrading the 2nd one we are passing argument "-Xmx512m" while starting the server and asking to use only 512 MB but is it is taking more than 1GB.

So I just want to whether we have any argument to reduce the memory utilization?

**Comment by kaisalmen** [comment]
> very quickly less than 4 seconds.

That is already long. First feedback or every time? Did you ever try without a docker container. What is your host OS? 

> Currently we are using [this](https://github.com/eclipse/eclipse.jdt.ls) for Language Server

Yes, we are talking about the same thing jdt.ls = eclipse.jdt.ls
 "-Xmx512m" configures the maxiumum heap. A Java process consumes other memory as well. With `-Xms` and `-Xmx` you configure only the minimum or maximum amount of heap memory space. If you set both to the same value (e.g. `-Xms1G -Xmx1G` you ensure heap is constantly available and not returned to the system.
The jdt.ls README states 1G of maximum heap (`-Xmx1G`), so only supplying half as maximum could already be problematic.

You see, so far this has nothing to do with monaco-languageclient, but now you have some more leads to follow hopefully leading to a resolution of your problems.

**Reply by LavanyaBurlagadda** [reply]
`That is already long. First feedback or every time? Did you ever try without a docker container. What is your host OS?`

For the first feedback, once language server is connected we are getting instantly in local server.

OS - Ubuntu
